submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/submitter/SubmissionSubmitter.kt

```diff
@@ -17,10 +22,39 @@ class SubmissionSubmitter(
         submission: ExtendedSubmission,
         files: FilesSource,
         context: PersistenceContext
+
     ): Submission {
-        validators.forEach { validator -> validator.validate(submission, context) }
-        processors.forEach { processor -> processor.process(submission, context) }
-        filesHandler.processFiles(submission, files)
+        val validationNodes = mutableListOf<ValidationNode>()
+
+        validators.map {
+            try {
+                it.validate(submission, context)
+            } catch (e: ValidationException) {
+                ValidationNode(ValidationNodeStatus.ERROR, e.message ?: e.javaClass.name)
+            }
+        }.filterIsInstance<ValidationNode>().ifNotEmpty {
+            validationNodes.add(ValidationNode(ValidationNodeStatus.ERROR, "Validation errors", it))
+        }
+
+        processors.map {
+            try {
+                it.process(submission, context)
+            } catch (e: ValidationException) {
+                ValidationNode(ValidationNodeStatus.ERROR, e.message ?: e.javaClass.name)
+            }
+        }.filterIsInstance<ValidationNode>().ifNotEmpty {
+            validationNodes.add(ValidationNode(ValidationNodeStatus.ERROR, "Processing errors", it))
+        }
+
+        try {
```

Comment:
Then, verifying whether or not this context object contains errors and handle them properly.